src/client/profile-provider.ts

```diff
@@ -627,39 +562,43 @@ export class ProfileProvider {
     input: T | string | undefined,
     parseFile: (contents: string, fileName?: string) => Promise<T>,
     unpackNested: (input: string) => T | string | undefined,
-    extensions: string[] = ['']
+    fileSystem: IFileSystem,
+    extensions: string[] = [''],
+    log?: LogFunction
   ): Promise<T | undefined> {
     if (typeof input === 'string') {
       if (isFileURIString(input)) {
         const fileName = input.slice(FILE_URI_PROTOCOL.length);
-        profileProviderDebug('Resolving input as file:', fileName);
+        log?.('Resolving input as file:', fileName);
 
         // read in files
         let contents, fileNameWithExtension;
         for (const extension of extensions) {
           fileNameWithExtension = fileName + extension;
-          try {
-            contents = await fsp.readFile(fileNameWithExtension, {
-              encoding: 'utf-8',
-            });
+          contents = await fileSystem.readFile(fileNameWithExtension);
+          if (contents.isOk()) {
             break;
-          } catch (e) {
-            void e;
           }
         }
 
-        if (contents === undefined) {
+        if (contents === undefined || contents.isErr()) {
```

Comment:
We were `void e` before now we throw `referencedFileNotFoundError`. Not sure if this change is intentional and if we really want to throw referencedFileNotFoundError 🤷🏻‍♂️

Reply:
read it again carefully, the behaviour did not change